ui/src/shared/utils/useVisDomainSettings.ts

```diff
@@ -54,3 +54,42 @@ export const useVisDomainSettings = (
 
   return [domain, setDomain, resetDomain]
 }
+
+export const getRemainingRange = (
+  data: NumericColumnData = [],
+  timeRange: TimeRange | null,
+  storedDomain: number[]
+) => {
+  const range = extent((data as number[]) || [])
+  if (range && range.length >= 2) {
+    const startTime = getStartTime(timeRange)
+    const endTime = getEndTime(timeRange)
+    const start = storedDomain[0]
+      ? storedDomain[0]
+      : Math.min(startTime, range[0])
+    const end = storedDomain[1] ? storedDomain[1] : Math.max(endTime, range[1])
+    return [start, end]
+  }
+  return range
+}
+
+export const useVisYDomainSettings = (
```

Comment:
Ya, I thought about this (and parseBounds) but wasn't positive that the useVisDomainSettings was being used in places that weren't entirely x-axis related so I didn't think it warranted updating the existing name to add more specificity. As for the `userVisYDomainsettings`, i wanted to maintain as close a relationship to the original variable name while explicating on what the difference is. In this name, I added the `Y` as a means of differentiating that this function only handles y-axis Domain settings

Reply:
After further investigation, it seems like we can specific `useVisDomainSettings` to `useVisXAxisDomainSettings` since that's the only context it's being used